### PR TITLE
Sync profile info and menu #37 & #72

### LINKED ISSUE #37: Display FxA profile (Management UI)
Child of #38

As a user, I want to know that I am signed in with Firefox Account (FxA).

### Acceptance Criteria
* Should keep the "Logins" tab, but not display the "Monitor" tab for now
* Should render FxA information if we have it, otherwise we need to show a 'guest' state if you're not signed in (need follow up stories for how we promote people to sign in)

### Engineering Notes

* This issue includes updating/modifying the "header" to match the current tabs-like visuals.
* The avatar image does **NOT** include the circle border (URL links to a "square" avatar)
* The `avatarURL` property is always set (using the default if the user has not uploaded one)
* Use `displayName` if set, otherwise `email`

### Visual Design
**Zeplin**
https://zpl.io/boK9ggk (current, may change depending on experiment)

### LINKED ISSUE #72: Implement Sync/Profile user menu
Child of #38

## Acceptance criteria

* Clicking on display name/email or avatar should open menu
* Clicking outside of menu should dismiss it
* ESC key should dismiss it

## Visual Design
**InVision**
Signed In: https://mozilla.invisionapp.com/share/ENOL84959AK#/338158451_Logins_-_Home_-_Menu_-Light-
Not Signed In: https://mozilla.invisionapp.com/share/ENOL84959AK#/339623981_Logins_-_Home_-_Menu_-_Not_Signed_In_-Light-
Hover: https://mozilla.invisionapp.com/share/ENOL84959AK#/344286028_Logins_-_Home_-_Menu_-_Hover_-Light-

**Zeplin**
Signed In: https://zpl.io/aRWJ48g
Not Signed In: https://zpl.io/a8pvROX
Hover: https://zpl.io/VQKLLYA

## Implementation Notes
* **FAQ Link:** https://lockbox.firefox.com/faq.html
* **Feedback form:** _TBD_

### PR DESCRIPTION
**Pull request by lmorchard**
Working toward implementing the profile menu and info display. Running into some issues and looking for an image asset, but figured I'd get a PR started for general kibbitzing.

Only meant to tackle one of these, but the two seemed to naturally run into each other.

Fixes #37
Fixes #72 

## Testing and Review Notes

- Profile info in upper right should be reactive to signing in and out of sync.
- Clicking on the icon in upper right should make a menu appear, which is adaptive to sync sign-in status.

## To Do

- [ ] Add unit tests and get coverage back up
- [x] Wrap en-US strings for l10n
- [x] Handle in Issue #92 ~Figure out how to link to "Sign in to Sync" - just opening an `about:sync` URL is not allowed. Maybe needs a Sync API experiment addition?~
- [x] Need to find a "pointer" image to add to the menu bubble so it's connected to the avatar visually
- [x] Handle in other issues (e.g.) ~Need to verify other menu items are linking to the right things~
- [x] Ensure the avatar display is correct - my account has no avatar, so it's just the anonymous image.
- [x] Remove the dev hack commit that opens the management page on reload!

### CODE REVIEW

**Comment by linuxwolf** [comment]
>  Figure out how to link to "Sign in to Sync" - just opening an about:sync URL is not allowed. Maybe needs a Sync API experiment addition?

#61 has some details on this, but the "TL;DR" is we'll need an extra method on the `sync` API to make the right calls and open the page.  For some reason, I thought we already had, but now seem to recall from IRL discussions it would be pushed until this issue was to get done (-:

**Comment by 6a68** [comment]
@lmorchard The sync folks pointed me at `gSync.openPrefs` as the right way to open the about:preferences#sync page, which we would indeed need to call from a new Sync API function: https://searchfox.org/mozilla-central/rev/c21d6620d384dfb13ede6054015da05a6353b899/browser/base/content/browser-sync.js#656

See also the usage of openPrefs elsewhere in that file: https://searchfox.org/mozilla-central/rev/c21d6620d384dfb13ede6054015da05a6353b899/browser/base/content/browser-sync.js#273

I think we want to change the `entryPoint` value to something lockbox-specific. I've been told this will be passed to fxa server-side telemetry, but not firefox telemetry, so it'll help tell us something indirectly about lockbox usage, but I'm not sure where we'd put that in the metrics.md doc. As for addon telemetry, the metrics doc does mention `reconnect_sync` and `signin_sync` for clicks on this menu item.

**Comment by 6a68** [comment]
whoops! I gotta remember to refresh before leaving comments :-P

**Comment by linuxwolf** [comment]
We ought to discuss/verify with @mozilla-lockbox/product ...

I believe (but seem to not have recorded herein) that handing some specific calls -- like opening the `sync` prefs a la #61 -- is deferred to those issues.  The rationale being to divide the work into accomplishable chunks.

Personally, I would consider making making the individual menu items truly work to be a stretch goal for this PR, and should be captured in other issues already.

**Comment by lmorchard** [comment]
> Personally, I would consider making making the individual menu items truly work to be a stretch goal for this PR, and should be captured in other issues already.

Just noticed issue #92 - I'd be fine with deferring the menu items here to other issues. I'm kind of guessing on some of the URLs, and I assume there should / will be telemetry asks for them that I haven't implemented here.

**Comment by linuxwolf** [comment]
> Just noticed issue #92 - I'd be fine with deferring the menu items here to other issues. I'm kind of guessing on some of the URLs, and I assume there should / will be telemetry asks for them that I haven't implemented here.

In your defense, some of the expectations/details weren't as spelled out as maybe they should have been before you started the work.

I'm walking through this PR to leave comments, hopefully it helps avoid some extra churn.

File: src/list/actions.js [hunk 1]
```diff
@@ -331,3 +344,47 @@ export function openFAQ() {
     type: OPEN_FAQ,
   };
 }
+
+export function openSyncPrefs() {
+  window.open(SYNC_PREFS_URL, "_blank");
```
Comment (linuxwolf) [inline src/list/actions.js]: as noted in IRL and chats, this needs to be an API call on `browser.experiments.sync` ... can be deferred to #92 / #61

File: src/list/actions.js [hunk 2]
```diff
@@ -331,3 +344,47 @@ export function openFAQ() {
...
+  return { type: OPEN_SYNC_PREFS };
+}
+
+export function openAccount() {
+  window.open(ACCOUNT_URL, "_blank");
```
Comment (linuxwolf) [inline src/list/actions.js]: never explicitly noted in the issue; this actually needs to go to sync prefs

File: src/list/actions.js [hunk 3]
```diff
@@ -331,3 +344,47 @@ export function openFAQ() {
...
+  return { type: OPEN_ACCOUNT };
+}
+
+export function openConnectDevice() {
+  window.open(CONNECT_DEVICE_URL, "_blank");
```
Comment (linuxwolf) [inline src/list/actions.js]: this is to be an in-content dialog; addressed in issue #TBD 😄 (though there are [designs](https://mozilla.invisionapp.com/share/ENOL84959AK#/screens/342439032))

File: src/list/actions.js [hunk 4]
```diff
@@ -36,13 +36,26 @@ export const HIDE_MODAL = Symbol("HIDE_MODAL");
...
+export const GET_PROFILE = Symbol("GET_PROFILE");
+export const GET_PROFILE_STARTING = Symbol("GET_PROFILE_STARTING");
+export const UPDATED_PROFILE = Symbol("UPDATED_PROFILE");
 
 // The action ID is used for debugging to correlate async actions with each
 // other (i.e. FOO_STARTING and FOO_COMPLETED).
 let nextActionId = 0;
 
 const FEEDBACK_URL = "https://qsurvey.mozilla.com/s3/Lockbox-Input?ver=" + version;
 const FAQ_URL = "https://mozilla-lockbox.github.io/lockbox-extension/faqs/";
```
Comment (linuxwolf) [inline src/list/actions.js]: One of the intents of #72 was to update the FAQ an Feedback links -- the issue has the correct link for the FAQ, but we don't yet have the correct Feedback link (will have a followup)

**Comment by lmorchard** [comment]
Whew, okay - this should have unit tests now, except for some of the TODO bits.

**Pull request merged**